### DISCUSSION TITLE
how to change concentration units

### DISCUSSION THREAD

**Discussion by fcobarahona** [Open-Systems-Pharmacology/Forum#1092] [Questions & Problems]
Hello, I was wondering how I can change the concentration units in which the individual data of a population simulation are exported, I need them in mg/l and they are exported in umol/l

**Comment by StephanSchaller** [comment]
You have to change the Y-axis unit to the desired unit before exporting the results:
1. Change Unit by right-clicking on the y-axis unit: 
![image](https://user-images.githubusercontent.com/25477643/151124247-6e2171ed-a009-45df-9a5e-4255a38611ec.png)

2. Change Unit in "Curves and Axis Options": 
![image](https://user-images.githubusercontent.com/25477643/151124311-eabc0d18-d6f7-439a-8eeb-4eedaf601f46.png)

3. Export results: 
![image](https://user-images.githubusercontent.com/25477643/151124343-0ea26472-4a0f-4c45-aa55-941b92cdb9cc.png)

**Reply by fcobarahona** [reply]
Hello, thank you very much, I think I did not explain myself well. what I need to change are the concentrations of the individual data obtained for a population simulation as shown here
![imagen](https://user-images.githubusercontent.com/85248912/151125508-398cc4cf-04f5-40a4-8eda-ade3c1dc11fd.png)

**Reply by fcobarahona** [reply]
additionally these are results displayed as molar concentration, i was wondering if they could just be displayed as concentration

**Reply by finattig** [reply]
I am having the same problem. Even though I set the simulation to the desired unit, run the simulation, and the plot shows the concentrations in the configured unit, all CSV outputs that have simulated concentration measurements are exported in umol/L. I also tried changing the display units options in Utilities, but that didn't work either.

**Reply by msevestre** [reply]
@fcobarahona @finattig and everyone else

The CSV is always exported in the internal unit which is umol/l

This is by design.  The goal is that other tools such as our R toolbox can load the results without having to do any conversion. (think molar vs mass concentration, it would be impossible to import results without knowing the MW of the underlying compound)

Should we change this? Probably not. 

However, it feels like there is a need to export the value in the desired DISPLAY unit. This would probably have to be another option in the software. @Yuri05 @StephanSchaller thoughts?

**Comment by PavelBal** [comment]
> However, it feels like there is a need to export the value in the desired DISPLAY unit. This would probably have to be another option in the software.

Maybe we should UNIFY the export functionalities.

- For individual simulations in PK-Sim, we have export to .csv and to Excel - not obvious what is the difference:
![image](https://user-images.githubusercontent.com/25903040/201992604-392eeabb-f398-4a27-8f80-c6495323c79d.png)

- For population simulations, only export to .csv:
![image](https://user-images.githubusercontent.com/25903040/201992714-ea89c269-f1fc-474f-b319-1cd7b1afdb78.png)

- In MoBi, only export to xls:

![image](https://user-images.githubusercontent.com/25903040/201992820-07577425-f783-48f0-9787-147b545f95ee.png)

**Reply by msevestre** [reply]
Indeed
